File: best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.go
func maxProfit(prices []int) int {
    if len(prices) < 2{
        return 0
    }
    maxprofit := 0
    for i:=0;i<len(prices)-1;i++{
        if prices[i+1] > prices[i]{
            maxprofit += prices[i+1] - prices[i] 
        }
    }
    return maxprofit
}

File: best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        totalprofit = 0
        for i in range(len(prices)-1):
            if prices[i] < prices[i+1]:
                totalprofit += prices[i+1] - prices[i]
        return totalprofit